pkg/remotewrite/remotewrite.go

```diff
@@ -41,16 +42,24 @@ func New(params output.Params) (*Output, error) {
 		return nil, err
 	}
 
+	var tm trendMapper
```

Comment:
If I got the code properly, that also can be a part of the separation (I mean that particular can be part of the sink structures) that I mentioned an in another comment and somewhere bellow you can call somewhere bellow: 

instead of
```golang
newts = swm.TrendMapper.Map(swm.TimeSeries, swm.Latest, trend)
```
Like that:
```
newts = trend.Map(swm.TimeSeries, swm.Latest)
```

Reply:
Thinking better on this, I think this must be refactored. The current design generates a mapper for each series. We should try reversing the pattern, we should allocate one mapper and pass seriesWithMeasure to it. I will try to change it.

Reply:
> The current design generates a mapper for each series. We should try reversing the pattern, we should allocate one mapper and pass seriesWithMeasure to it

I'm not sure if I got this :thinking: Is the mapper itself a heavy struct?

Just in case, I see no issues with having different mappers. It's absolutely okay when we have complex data structures.

What I'm saying is that in the case of that PR `Map` function belongs to the sink's structs. Basically, in that case, sync will just satisfy the `Mapper` interface, and we don't need somewhere also keep the correct reference as we do now `output.trendMapper`